### PR TITLE
refactor: Remove Optional input parameters

### PR DESCRIPTION
**Pull request by jeqo**
Cleanup code accepting Optional types, see https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/531#discussion_r1553816416

### CODE REVIEW

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java [hunk 1]
```diff
@@ -60,11 +56,11 @@ private static ConfigDef addCacheConfigs(final OptionalLong maybeDefaultSize) {
...
 
     public CacheConfig(final Map<String, ?> props) {
-        super(addCacheConfigs(OptionalLong.empty()), props);
+        super(configDef(NO_DEFAULT_VALUE), props);
     }
 
-    public CacheConfig(final Map<String, ?> props, final long defaultSize) {
-        super(addCacheConfigs(OptionalLong.of(defaultSize)), props);
+    public CacheConfig(final Map<String, ?> props, final Long defaultSize) {
+        super(configDef(Objects.requireNonNull(defaultSize, "defaultSize cannot be null")), props);
```
Comment (ivanyu) [inline core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java:63]: If it can't be `null`, should we keep `long` as the type here?

**Review by ivanyu — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java (hunk 1 above)
Comment (jeqo) [reply core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java]: Good catch. Added a Builder to conditionally pass this value when needed. PTAL.

File: core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java [hunk 2]
```diff
@@ -82,4 +77,22 @@ public Optional<Duration> cacheRetention() {
         }
         return Optional.of(Duration.ofMillis(rawValue));
     }
+
+    public static class Builder {
+        final Map<String, ?> props;
+        Object maybeDefaultSize = NO_DEFAULT_VALUE;
```
Comment (ivanyu) [inline core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java:83]:
```suggestion
        private final Map<String, ?> props;
        private Object maybeDefaultSize = NO_DEFAULT_VALUE;
```
I think

**Review by ivanyu — CHANGES_REQUESTED** [review]
LGTM, one small nit

**Review by ivanyu — APPROVED** [review]

**Pull request merged**